lib/matplotlib/tests/test_gridspec.py

```diff
@@ -27,6 +39,22 @@ def test_height_ratios():
         gridspec.GridSpec(1, 1, height_ratios=[2, 1, 3])
 
 
+def test_SubplotParams():
+    s = gridspec.SubplotParams(.1, .1, .9, .9)
+    assert s.left == 0.1
+
+    with pytest.raises(ValueError):
+        s.update(left=s.right + .01)
+
+    with pytest.raises(ValueError):
+        s.update(bottom=s.top + .01)
+
+    with pytest.raises(ValueError):
+        s = gridspec.SubplotParams(.1, .1, .09, .9)
+
+    s.reset()
+    assert s.left == matplotlib.rcParams['figure.subplot.left']
```

Comment:
Move this up before the failure tests, because it relies on an initially modified `left` to be meaningful. It's harder to reason about the state of subplotparams after the failure tests.